### DISCUSSION TITLE
How to define subdomains for a host that is resolved from a conditional forwarded zone?

### DISCUSSION THREAD

**Discussion by mpeter50** [TechnitiumSoftware/DnsServer#818] [Q&A] [answered]
Hi!

I would start with a few details of my existing setup.
My Technitium DNS server has access to a broader network that is served by a DHCP server running on a router device. The DHCP server is dnsmasq.
On this network there are hosts with hostnames like hosta and hostb, and the DHCP server is set up so that all hosts are on the ".lan" domain. With a preexisting DNS server which was set up for conditional forwarding, "hosta.lan" correctly resolves to the IP address currently given out by the DHCP server for this host.

In Technitium DNS I have already set up a new conditional forwarder zone for "lan" , with forwarder being set as the router's IP address. When I want to resolve "hosta.lan" with this DNS server, that is working ok like with the old one, I receive hosta's IP address.

However I have run into a problem with setting up subdomains for hosta in Technitium DNS. I have tried it in 2 ways.

First I have created a new primary zone, "hosta.lan". Before adding an A record for a subdomain to it, I wanted to test if this still works.
However now sending a query for hosta.lan does not return an A record, it only returns the SOA record of the zone. I guess this is because the hosta.lan zone does not have an A record, and the .lan zone is not searched for one.
I have tried adding a CNAME/DNAME record for the "@" name that points to "hosta.lan", in the hopes that it will search for records in the "lan" zone, but that did not work.

Then I have deleted this hosta.lan zone and tried to add my subdomain records to the lan zone.
There I have added a new A record for the "test1.hosta.lan" name with an IP address, and when I have sent a request for this domain I received a response with NXDOMAIN status.
Looking at the query log, which has logged this request with the "Recursive" response type, it seems as if the request would have been forwarded to my router, even though there was an exactly matching name in the zone.

What am I doing wrong? How should I set subdomains for hosts whose name is managed by an external DNS server? (thr router's DHCP server)
I don't have a lot of experience yet with DNS servers. I'm transitioning from Pi-hole, where I needed to add every single subdomain as a different record with the same IP, and my end goal is to be able to solve it with wildcard records for subdomains that work the same, and references to other records (like test.hosta.lan to hosta.lan) so that I don't have to hard code IP addresses and update them by hand if they change.
If possible I would like to make a separate zone for each host.

**Comment by ShreyasZare** [comment, marked as answer]
Thanks for the post. You should just have one conditional forwarder zone for "lan". You do not need to create a primary zone for the host, in fact you should not create for this scenario. Just add an A record with "test1.hosta" as the name and it will work.

In forwarder zones, if a record does not exists, it would forward the request to server specified in the FWD record. If you create a primary zone, its authoritative and it will not forward request but will respond with NXDOMAIN.

If you still have issue with the setup, then share screenshot of your forwarder zone either here or send it by email to support@technitium.com.

**Reply by mpeter50** [reply]
Ok, so I should stick to the .lan zone for this.

> Just add an A record with "test1.hosta" as the name and it will work.

Oh, so the trick is that when adding records to a zone, the name shouldn't contain the zone's name!
This way querying "test1.hosta.lan" returns the correct IP, even if the record is a CNAME targeting hosta.lan, or when the name contains a wildcard. Thanks!

**Reply by ShreyasZare** [reply]
> Oh, so the trick is that when adding records to a zone, the name shouldn't contain the zone's name!

Yes, you just need to enter the subdomain name part. The GUI will automatically construct the full domain name by appending the zone name to it.
